#referenc: http://bril-tech.blogspot.com/2016/10/ros4.html

armed3w:
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  diff_drive_controller:
    type        : "diff_drive_controller/DiffDriveController"
    left_wheel  : 'body_wheelLeft'
    right_wheel : 'body_wheelRight'
    publish_rate: 50.0               # default: 50
    pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]

    # Wheel separation and diameter. These are both optional.
    # diff_drive_controller will attempt to read either one or both from the
    # URDF if not specified as a parameter
    wheel_separation : 0.3464
    wheel_radius : 0.05

    # Wheel separation and radius multipliers
    wheel_separation_multiplier: 1.0 # default: 1.0
    wheel_radius_multiplier    : 1.0 # default: 1.0

    # Velocity commands timeout [s], default 0.5
    cmd_vel_timeout: 2.0
    
    # Base frame_id
    base_frame_id: base_footprint #default: base_link
    
    # Velocity and acceleration limits
    # Whenever a min_* is unspecified, default to -max_*
    linear:
      x:
        has_velocity_limits    : true
        max_velocity           : 0.5  # m/s
        min_velocity           : -0.5 # m/s
        has_acceleration_limits: true
        max_acceleration       : 0.5  # m/s^2
        min_acceleration       : -0.5 # m/s^2
    angular:
      y:
        has_velocity_limits    : false
        max_velocity           : 0.3  # rad/s
        min_velocity           : -0.3
        has_acceleration_limits: false
        max_acceleration       : 0.3  # rad/s^2
        min_acceleration       : -0.3

  arm_controller:
    type        : "diff_drive_controller/DiffDriveController"
    left_wheel  : 'body_armLeft'
    right_wheel : 'body_armRight'
    publish_rate: 50.0               # default: 50
    pose_covariance_diagonal : [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    twist_covariance_diagonal: [0.001, 0.001, 1000000.0, 1000000.0, 1000000.0, 1000.0]
    wheel_separation : 0.6
    wheel_radius : 0.63661
    cmd_vel_timeout: 4

